torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -375,7 +375,18 @@ void ProcessGroupAgent::enqueueRecv(RecvWork work) {
             work.type_,
             work.id_);
         if (message.isRequest()) {
-          send(work.from_, cb_->operator()(message));
+          auto futureResponse = cb_->operator()(message);
+          if (futureResponse->completed()) {
+            send(work.from_, std::move(*futureResponse).moveMessage());
+          } else {
+            auto fromId = work.from_.id_;
```

Comment:
Would I be correct if I assume current this path is not covered by any test/code?

Reply:
It's covered by caffe2/test:dist_autograd_spawn  (if I add add a LOG(FATAL), it's reached there)

I implemented the one async autograd op partly to get test coverage..